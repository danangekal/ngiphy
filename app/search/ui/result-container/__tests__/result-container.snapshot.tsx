import { useSearchParams } from 'next/navigation';
import { render } from '@testing-library/react';

import ResultContainer from '../index';

const mockAction = jest.fn();
jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      push: mockAction,
    };
  },
  useSearchParams: jest.fn(),
}));

it('Result container unchanged', () => {
  const mockGet = jest.fn();
  mockGet.mockReturnValue({ page: 1, rating: 'g', keyword: 'animals' });

  (useSearchParams as jest.Mock).mockReturnValue({
    get: mockGet,
  });

  const { container } = render(<ResultContainer />);
  expect(container).toMatchSnapshot();
});
