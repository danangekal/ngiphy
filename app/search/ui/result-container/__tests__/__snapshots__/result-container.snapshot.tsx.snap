// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result container unchanged 1`] = `
<div>
  <div
    class="flex flex-col w-full items-center justify-between"
  >
    <div
      class="flex flex-row items-start py-4 text-sm text-blue-500 "
    >
      <p>
        Search Result
      </p>
    </div>
    <div
      class="flex flex-col items-center justify-between w-full"
    >
      <p
        class="text-red-500 text-md"
      >
        Something Wrong!
      </p>
    </div>
  </div>
</div>
`;
